app/code/Magento/Store/Console/Command/StoreListCommand.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace Magento\Store\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+
+/**
+ * Class StoreListCommand
+ *
+ * Command for listing the configured stores
+ */
+class StoreListCommand extends Command
+{
+    /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
+    private $storeManager;
+
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->storeManager = $storeManager;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('store:list')
+            ->setDescription('Displays the list of stores');
+
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $table = $this->getHelperSet()->get('table');
```

Comment:
Please wrap the logic in `try-catch` and specify return codes for success and failure explicitly. See an example in https://github.com/magento/magento2/blob/develop/app/code/Magento/Deploy/Console/Command/SetModeCommand.php#L108
The same for another test